code/datums/storage/storage.dm

```diff
@@ -22,10 +31,15 @@
 	/// List of all the mobs currently viewing the contents of this storage.
 	VAR_PRIVATE/list/mob/is_using = list()
 
-	/// The storage display screen object.
-	VAR_PRIVATE/atom/movable/screen/storage/boxes
-	/// The 'close button' screen object.
-	VAR_PRIVATE/atom/movable/screen/close/closer
+	/// Associated list of all storage cells and corners.
+	/// In order to minimize memory taken, only one storage UI set is generated per UI theme.
+	VAR_PRIVATE/list/storage_sets = list()
+	/// Associated list of all "close storage" buttons.
+	VAR_PRIVATE/list/atom/movable/screen/close/storage_closers = list()
+	/// Associated list that tracks how many mobs are viewing this storage with a certain UI theme. Used to delete unused UI theme objects.
+	VAR_PRIVATE/list/storage_themes = list()
```

Comment:
This seems ill advised. is the problem that we need to generate a unique theme object for every viewer? what do the theme objects do?

Reply:
We need to generate a set of elements for every single theme that is currently in use. I.e. if two people with Midnight theme open a box, it only generates one set (like currently on live). This is just an associated list that keeps track of how many people with a certain theme are viewing the storage, so it knows when to dispose of unneeded elements.